c/cert/test/rules/FIO47-C/test.c

```diff
@@ -0,0 +1,544 @@
+
+  printf("%hg", v_double);       // NON_COMPLIANT
+  printf("%hhg", v_double);      // NON_COMPLIANT
+  printf("%lg", v_double);       // COMPLIANT
+  printf("%llg", v_long_double); // COMPLIANT
+  printf("%jg", v_double);       // NON_COMPLIANT
+  printf("%zg", v_double);       // NON_COMPLIANT
+  printf("%tg", v_double);       // NON_COMPLIANT
+  printf("%Lg", v_long_double);  // COMPLIANT
+
+  printf("%hG", v_double);       // NON_COMPLIANT
+  printf("%hhG", v_double);      // NON_COMPLIANT
+  printf("%lG", v_double);       // COMPLIANT
+  printf("%llG", v_long_double); // COMPLIANT
+  printf("%jG", v_double);       // NON_COMPLIANT
+  printf("%zG", v_double);       // NON_COMPLIANT
+  printf("%tG", v_double);       // NON_COMPLIANT
+  printf("%LG", v_long_double);  // COMPLIANT
+
+  printf("%ha", v_double);       // NON_COMPLIANT
+  printf("%hha", v_double);      // NON_COMPLIANT
+  printf("%la", v_double);       // COMPLIANT
+  printf("%lla", v_long_double); // COMPLIANT
+  printf("%ja", v_double);       // NON_COMPLIANT
+  printf("%za", v_double);       // NON_COMPLIANT
+  printf("%ta", v_double);       // NON_COMPLIANT
+  printf("%La", v_long_double);  // COMPLIANT
+
+  printf("%hA", v_double);       // NON_COMPLIANT
+  printf("%hhA", v_double);      // NON_COMPLIANT
+  printf("%lA", v_double);       // COMPLIANT
+  printf("%llA", v_long_double); // COMPLIANT
+  printf("%jA", v_double);       // NON_COMPLIANT
+  printf("%zA", v_double);       // NON_COMPLIANT
+  printf("%tA", v_double);       // NON_COMPLIANT
+  printf("%LA", v_long_double);  // COMPLIANT
+
+  printf("%hc", v_int);    // NON_COMPLIANT
+  printf("%hhc", v_int);   // NON_COMPLIANT
+  printf("%lc", v_wint_t); // COMPLIANT
+  printf("%llc", v_int);   // NON_COMPLIANT
+  printf("%jc", v_int);    // NON_COMPLIANT
+  printf("%zc", v_int);    // NON_COMPLIANT
+  printf("%tc", v_int);    // NON_COMPLIANT
+  printf("%Lc", v_int);    // NON_COMPLIANT
+
+  printf("%hs", v_char_ptr);    // NON_COMPLIANT
+  printf("%hhs", v_char_ptr);   // NON_COMPLIANT
+  printf("%ls", v_wchar_t_ptr); // COMPLIANT
+  printf("%lls", v_char_ptr);   // NON_COMPLIANT
+  printf("%js", v_char_ptr);    // NON_COMPLIANT
+  printf("%zs", v_char_ptr);    // NON_COMPLIANT
+  printf("%ts", v_char_ptr);    // NON_COMPLIANT
+  printf("%Ls", v_char_ptr);    // NON_COMPLIANT
+
+  printf("%hp", v_void_ptr);  // NON_COMPLIANT
+  printf("%hhp", v_void_ptr); // NON_COMPLIANT
+  printf("%lp", v_void_ptr);  // NON_COMPLIANT
+  printf("%llp", v_void_ptr); // NON_COMPLIANT
+  printf("%jp", v_void_ptr);  // NON_COMPLIANT
+  printf("%zp", v_void_ptr);  // NON_COMPLIANT
+  printf("%tp", v_void_ptr);  // NON_COMPLIANT
+  printf("%Lp", v_void_ptr);  // NON_COMPLIANT
+
+  printf("%hn", v_short_ptr);      // COMPLIANT
+  printf("%hhn", v_char_ptr);      // COMPLIANT
+  printf("%ln", v_long_ptr);       // COMPLIANT
+  printf("%lln", v_long_long_ptr); // COMPLIANT
+  printf("%jn", v_intmax_t_ptr);   // COMPLIANT[FALSE_POSITIVE]
```

Comment:
not sure if we want to annotate which part of the rule is FP'ing on these to make the nuances of the rule behaviour more obvious when just looking at the testcase? unless it already is, but I spent a minute trying to see why `jn` was reported as an invalid combination before realizing thats not where it was coming from 😅

Reply:
"Do you mean putting the name of the imprecise query?" - yeah , usually its more obvious I think with less testcases but in this file it might be easier to see the query limitations with those specifically annotated. just a suggestion though